robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowImageView.java

```diff
@@ -34,7 +34,11 @@ public Bitmap getImageBitmap() {
 
   @Implementation
   public void setImageDrawable(Drawable drawable) {
-    this.imageDrawable = drawable;
+
+    if(this.imageDrawable != drawable) {
+      this.imageDrawable = drawable;
+      invalidate();
```

Comment:
Are you intending to call `invalidate` on the real object? If so, you need to use `directlyOn(realObject, "invalidate")`. Otherwise the method call will get re-routed back to the shadow first.